core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -721,6 +725,7 @@ class KafkaApis(val requestChannel: RequestChannel,
 
     if (interesting.isEmpty) {
       processResponseCallback(Seq.empty)
+      CompletableFuture.completedFuture[Unit](())
```

Comment:
There is `CompletableFuture.completedFuture[Unit](())` at the end of the function. We can remove this one.